CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -947,6 +948,16 @@ public static function processFormContribution(
 
         $form->_params['pledge_id'] = $pledge->id;
 
+        // Create the recur record.
+        $recurringPledgeID = CRM_Pledge_BAO_Pledge::createRecurRecord($pledge, $params, $form);
+        if ($recurringPledgeID) {
+          $contribParams = array(
+            'id' => $contribution->id,
+            'contribution_recur_id' => $recurringPledgeID,
+          );
+          CRM_Contribute_BAO_Contribution::add($contribParams);
```

Comment:
Let's use the api from the form layer as much as possible - there was a general feeling at CiviCon in favour of that